dencoding/yaml_decoder.go

```diff
@@ -72,20 +74,26 @@ func (decoder *YAMLDecoder) getNodeValue(node *yaml.Node) (any, error) {
 func (decoder *YAMLDecoder) getMappingNodeValue(node *yaml.Node) (any, error) {
 	res := NewMap()
 
-	for i := 0; i < len(node.Content); i += 2 {
-		keyNode := node.Content[i]
-		valueNode := node.Content[i+1]
+	content := make([]*yaml.Node, 0)
+	content = append(content, node.Content...)
+
+	var keyNode *yaml.Node
+	var valueNode *yaml.Node
+	for {
+		if len(content) == 0 {
+			break
+		}
+
+		keyNode, valueNode, content = content[0], content[1], content[2:]
```

Comment:
Since we need to merge the content of the alias into the current content of the map, I found it easiest to turn this into a queue rather than trying to manipulate the index.